app/src/org/commcare/models/database/SqlStorage.java

```diff
@@ -94,16 +103,23 @@ public SqlStorage(String table, Class<? extends T> ctype, AndroidDbHelper helper
         }
 
         Cursor c = db.query(table, new String[]{DatabaseHelper.ID_COL}, whereClause.first, whereClause.second, null, null, null);
-        return fillIdWindow(c, DatabaseHelper.ID_COL);
+        return fillIdWindow(c, DatabaseHelper.ID_COL, returnSet);
     }
 
     public static Vector<Integer> fillIdWindow(Cursor c, String columnName) {
+        return fillIdWindow(c, columnName, null);
+    }
+
+    public static Vector<Integer> fillIdWindow(Cursor c, String columnName, LinkedHashSet<Integer> newReturn) {
```

Comment:
This is a hack, essentially, that I needed to do to not end up rewriting a bunch of code to be based on LinkedHashSet instead of Vectors. 

I did a first pass that refactored everything to use collections, but the overall impact to the code was immense and there were some design discussions that didn't feel fair to couple with the PR.  It's technical debt on my part, but felt more efficient than forcing everyone to review detailed changes to the vector datatype while also reviewing the PR.